cwms-data-api/src/test/java/cwms/cda/api/TimeseriesControllerTestIT.java

```diff
@@ -1,32 +1,709 @@
-
 import cwms.cda.formatters.Formats;
 import fixtures.TestAccounts;
-import fixtures.TestAccounts.KeyUser;
-import io.restassured.RestAssured;
+import io.jsonwebtoken.io.IOException;
 import io.restassured.filter.log.LogDetail;
-import io.restassured.path.json.config.JsonPathConfig;
 import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
 
 import static io.restassured.RestAssured.given;
-import static io.restassured.config.JsonConfig.jsonConfig;
-import static org.hamcrest.Matchers.closeTo;
-import static org.hamcrest.Matchers.contains;
-import static org.hamcrest.Matchers.is;
-import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 
 @Tag("integration")
 public class TimeseriesControllerTestIT extends DataApiTestIT {
 
+    private static final String OFFICE = "SPK";
+    private static final String locationId = "TsVersionedTestLoc";
+    private static final String tsId = locationId + ".Flow.Inst.1Hour.0.raw";
+    public static final String BEGIN_STR = "2008-05-01T15:00:00Z";
+    public static final String END_STR = "2008-05-01T17:00:00Z";
+
+    @BeforeAll
+    public static void create() throws Exception {
+        createLocation(locationId, true, OFFICE);
+        createTimeseries(OFFICE, tsId, 0);
+    }
+
+    @Test
+    void test_create_get() throws IOException, java.io.IOException {
+        // Post versioned time series
+        InputStream resource = this.getClass().getResourceAsStream("/cwms/cda/api/spk/num_ts_create.json");
+        assertNotNull(resource);
+        String tsData = IOUtils.toString(resource, StandardCharsets.UTF_8);
+        assertNotNull(tsData);
+
+        TestAccounts.KeyUser user = TestAccounts.KeyUser.SPK_NORMAL;
+
+        given()
+            .log().ifValidationFails(LogDetail.ALL,true)
+            .accept(Formats.JSONV2)
+            .contentType(Formats.JSONV2)
+            .body(tsData)
+            .header("Authorization", user.toHeaderValue())
+        .when()
+            .redirects().follow(true)
+            .redirects().max(3)
+            .post("/timeseries/")
+        .then()
+            .log().ifValidationFails(LogDetail.ALL,true)
+            .assertThat()
+            .statusCode(is(HttpServletResponse.SC_OK));
+
+
+        // Get versioned time series
+        given()
+            .log().ifValidationFails(LogDetail.ALL,true)
+            .accept(Formats.JSONV2)
+            .contentType(Formats.JSONV2)
+            .header("Authorization", user.toHeaderValue())
+            .queryParam("office", OFFICE)
+            .queryParam("name", tsId)
+            .queryParam("begin", BEGIN_STR)
+            .queryParam("end", END_STR)
+            .queryParam("date-version-type", "SINGLE_VERSION")
+            .queryParam("version-date", "2021-06-21T08:00:00-0000[UTC]")
```

Comment:
Can you pull this from the deserialized `TimeSeries` object? That way it's only written once.

Reply:
Done. I didn't use a Timseries object because it required a lot of parameters I didn't have values for so I created a paramMap object I could use for the different requests

Reply:
Your file `num_ts_create.json` is a `TimeSeries` object, you just have to deserialize it

Reply:
Okay I have other parameters that change depending on the call. I could use the Timeseries object for some of the query params but others like end that change from query to query would have to be put in this way or I would need to create a setter for it

Reply:
done